grant/projects/offchain-subnames.md

```diff
@@ -0,0 +1,99 @@
+# Offchain Subnames on Darwinia Chain
+
+## Project Description
+
+This project aims to implement an offchain subnames solution for the 'ringdao.eth' Ethereum Name Service (ENS) domain but keeps the subname records on Darwinia Chain. 
+
+It leverages the EIP-3668 specification to enable efficient and cost-effective management of subname-address mappings off the Ethereum mainnet. By storing these mappings on Darwinia Chain, the project offers a scalable approach to handle a large number of subdomains without incurring high gas fees.
+
+The solution utilizes a customized OffchainResolver contract on Ethereum, which works with a gateway service connected to the Darwinia Chain. This setup allows for seamless resolution of subnames (e.g., 'foo.ringdao.eth') while maintaining the security and decentralization benefits of the ENS.
+
+<!-- Key features of the project include:
+1. Offchain storage of subname-address mappings on Darwinia Chain
+2. Integration with ENS on Ethereum using EIP-3668 for secure offchain data retrieval
+3. A gateway service to facilitate communication between Ethereum and Darwinia Chain
+4. User-friendly UIs for subname registration, update, and resolution.
+5. Admin UI for subname management. -->
+
+This project addresses the need for efficient subdomain management within the Darwinia ecosystem, as outlined in the RFP (https://github.com/ringecosystem/collaboration/issues/61), and demonstrates the potential for cross-chain interoperability in domain name services.
+
+### This project will benefit the Darwinia ecosystem in the following ways:
+
+1. It will provide a more efficient and cost-effective way to manage subdomains on the Darwinia Chain, reducing the need for users to pay high gas fees on Ethereum.
+2. It will enhance the usability of the 'ringdao.eth' domain by allowing users to register, update, and resolve subnames with reduced costs and improved performance.
+3. It will demonstrate the potential for cross-chain interoperability in domain name services, opening up new possibilities for collaboration and innovation within the Darwinia ecosystem.
+
+## Team
+
+- Aki Wu([wuminzhe](https://github.com/wuminzhe))  
+  A full stack developer focused on the web3 space. He has been working on darwinia ecosystem for a long time.
+
+## Team Code Repos
+
+- https://github.com/wuminzhe/basenames
+  The repo for the basenames contracts.
+
+- https://github.com/wuminzhe/offchain-resolver  
+  The repo provides the offchain resolve client.
+
+- https://github.com/wuminzhe/subnames-ui
+  The repo for the UI.
+
+- https://github.com/wuminzhe/subnames-gateway
+  The repo for the offchain gateway.
```

Comment:
These seems not open source. It's OK for now, but open source and license need to be committed in  delivery of each milestones.

Reply:
- https://github.com/wuminzhe/offchain-resolver  
- https://github.com/wuminzhe/subnames-gateway
@wuminzhe Do these two repos do the same thing, and can they be merged?

Reply:
yes, I will opensource all

Reply:
Hi, please add the open source commitment in your tasks/deliveries in the application.